src/ci/docker/x86_64-gnu-tools/checktools.sh

```diff
@@ -105,21 +114,11 @@ status_check() {
 
 status_check "submodule_changed"
 
-CHECK_NOT="$(readlink -f "$(dirname $0)/checkregression.py")"
 # This callback is called by `commit_toolstate_change`, see `repo.sh`.
 change_toolstate() {
-    # only update the history
-    if python2.7 "$CHECK_NOT" "$OS" "$TOOLSTATE_FILE" "_data/latest.json" changed; then
-        echo 'Toolstate is not changed. Not updating.'
-    else
-        if [ $SIX_WEEK_CYCLE -ge 35 ]; then
-            # Reject any regressions during the week before beta cutoff.
-            python2.7 "$CHECK_NOT" "$OS" "$TOOLSTATE_FILE" "_data/latest.json" regressed
-        fi
-        sed -i "1 a\\
+    sed -i "1 a\\
 $COMMIT\t$(cat "$TOOLSTATE_FILE")
 " "history/$OS.tsv"
```

Comment:
Why do we not even do the "changed" test any more? That had nothing to do with regressing or not.
If I read this correctly, this would commit a new toolstate to the repo even for commits that do not change any toolstate. Is this intentional? That would be useful for https://github.com/rust-lang/rust/issues/60301 but I was under the impression that we deliberately reduced traffic for that repo.

Reply:
Oh and also, doesn't this *remove* the check that tools do not regress in the beta week? We only run that check when submodules change now, instead of on all PRs, it seems.

Reply:
Hm, yeah, that might be right. To be honest I'm having a really hard time understanding this code - I thought we'd handled those cases up above but maybe not :)